kornia/geometry/transform/image_registrator.py

```diff
@@ -0,0 +1,263 @@
+from typing import Optional, Tuple, Type, Union
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import torch.optim as optim
+
+from kornia.geometry.conversions import convert_affinematrix_to_homography
+from kornia.geometry.transform.homography_warper import HomographyWarper
+from kornia.geometry.transform.imgwarp import angle_to_rotation_matrix
+from kornia.geometry.transform.pyramid import build_pyramid
+from kornia.testing import check_is_tensor
+
+__all__ = [
+    "ImageRegistrator",
+    "Homography",
+    "Similarity"]
+
+
+class Homography(nn.Module):
+    def __init__(self):
+        r"""Homography geometric model to be used together with ImageRegistrator
+        module for the optimization-based image
+        registration."""
+        super().__init__()
+        self.model = nn.Parameter(torch.Tensor(3, 3))
+        self.reset_model()
+
+    def __repr__(self) -> str:
+        return f'{self.__class__.__name__}({self.model})'
+
+    def reset_model(self):
+        """Initializes the model with identity transform."""
+        torch.nn.init.eye_(self.model)
+
+    def forward(self) -> torch.Tensor:
+        r"""Single-batch homography".
+
+        Return:
+            Homography.
+
+        Shape:
+            - Output: :math:`(1, 3, 3)`"""
+        return torch.unsqueeze(self.model / self.model[2, 2], dim=0)  # 1x3x3
+
+    def forward_inverse(self) -> torch.Tensor:
+        r"""Interted Single-batch homography".
+
+        Return:
+            Homography.
+
+        Shape:
+            - Output: :math:`(1, 3, 3)`"""
+        return torch.unsqueeze(torch.inverse(self.model), dim=0)
+
+
+class Similarity(nn.Module):
+    def __init__(self,
+                 rotation: bool = True,
+                 scale: bool = True,
+                 shift: bool = True):
+        """Similarity geometric model to be used together with ImageRegistrator module for the optimization-based
+        image registration.
+
+        Args:
+            rotation (bool): if True, the rotation is optimizable, else constant zero
+            scale (bool): if True, the scale is optimizable, else constant zero
+            shift (bool): if True, the shift is optimizable, else constant one
+        """
+        super().__init__()
+        if rotation:
+            self.rot = nn.Parameter(torch.Tensor(1))
```

Comment:
```suggestion
            self.rot = nn.Parameter(torch.tensor(1))
```
same below - not recommended to initialize with class Tensor name